lib/ReplaceOpenCLBuiltinPass.cpp

```diff
@@ -300,6 +302,81 @@ bool ReplaceOpenCLBuiltinPass::replaceAbs(Module &M) {
   return Changed;
 }
 
+bool ReplaceOpenCLBuiltinPass::replaceCopysign(Module &M) {
+  bool Changed = false;
+
+  const char *Names[] = {
+    "_Z8copysignff",
```

Comment:
What about double and half support?

Reply:
The code should just work in both cases. clspv needs some general work for double support, I've created #321. For half, well, there aren't any conformance tests (see https://github.com/KhronosGroup/OpenCL-CTS/issues/142). I suggest looking at half support/testing this as part of #276 or creating another issue similar to #321 as support is missing in quite a few places.